bool acknowledge_request
---
bool acknowledge_response